src/main/java/echoquery/sql/QueryBuilder.java

```diff
@@ -84,17 +92,11 @@ public static QueryBuilder of(Intent intent) {
     } else {
       aggregationFunc = QueryUtil.functionCall(
           SlotUtil.getAggregateFunction(aggregate),
-          QueryUtil.nameReference(aggregationColumn));
+          new QualifiedNameReference(
+              QualifiedName.of(from.wherePrefix(1), aggregationColumn)));
```

Comment:
yeah I totally agree, so what the index refers to is the order that the columns where passed in to infer. it totally is magic number-y i just wanted to get it up